apps/resource_app.go

```diff
@@ -0,0 +1,116 @@
+package apps
+
+import (
+	"context"
+	"log"
+	"time"
+
+	"github.com/databricks/databricks-sdk-go/service/apps"
+	"github.com/databricks/terraform-provider-databricks/common"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+const defaultAppProvisionTimeout = 10 * time.Minute
+const deleteCallTimeout = 10 * time.Second
+
+var appAliasMap = map[string]string{
+	"resources": "resource",
+}
+
+type appStruct struct {
+	apps.App
+}
+
+func (appStruct) Aliases() map[string]map[string]string {
+	return map[string]map[string]string{
+		"apps.appStruct": appAliasMap,
+	}
+}
+
+func (appStruct) CustomizeSchema(s *common.CustomizableSchema) *common.CustomizableSchema {
+
+	// Required fields & validation
+	s.SchemaPath("name").SetRequired().SetForceNew().SetValidateFunc(validation.StringLenBetween(2, 30))
+
+	// Computed fields
+	for _, p := range []string{"active_deployment", "app_status", "compute_status", "create_time", "creator",
+		"default_source_code_path", "pending_deployment", "service_principal_id", "service_principal_name",
+		"update_time", "updater", "url"} {
+		s.SchemaPath(p).SetComputed()
+	}
+	return s
```

Comment:
The resource types are mutually exclusive per resource element (`secret`, `sql_warehouse`, etc)

Reply:
you can add multiple resources to an app

Reply:
Yes, but within a resource you should only have one of (`secret`, `sql_warehouse`, etc).

Reply:
Could we add this constraint back, where you cannot have more than one defined per resource?

Reply:
because the type is nested within `resource`, we cannot add constraints with sdkv2 - would need to rewrite this to framework to be able to add this validation rule